src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -1031,9 +1057,59 @@ object FunctionRegistry {
     "nHomVar" -> "Number of samples that are homozygous for the alternate allele."
   )
 
+
+  def chisqTest(c1: Int, c2: Int, c3: Int, c4: Int): (Option[Double], Option[Double]) = {
+
+    var or: Option[Double] = None
+    var chisqp: Option[Double] = None
+
+    if (Array(c1, c2, c3, c4).sum > 0) {
+      if (c1 > 0 && c3 == 0)
+        or = Option(Double.PositiveInfinity)
+      else if (c3 > 0 && c1 == 0)
+        or = Option(Double.NegativeInfinity)
+      else if ((c1 > 0 && c2 > 0 && c3 > 0 && c4 > 0))
+        or = Option((c1 * c4) / (c2 * c3).toDouble)
+        chisqp = Option(chisq.chiSquareTest(Array(Array(c1, c2), Array(c3, c4))))
+    }
+
+    (chisqp, or)
+  }
+
+
+  val chisqStruct = TStruct(Array(("pValue", TDouble, "p-value"), ("oddsRatio", TDouble, "odds ratio")).zipWithIndex.map { case ((n, t, d), i) => Field(n, t, i, Map(("desc", d))) })
+  registerAnn("chisq", chisqStruct, { (c1: Int, c2: Int, c3: Int, c4: Int) =>
+    if (c1 < 0 || c2 < 0 || c3 < 0 || c4 < 0)
+      fatal(s"got invalid argument to function `chisq': chisq($c1, $c2, $c3, $c4)")
+
+    val res = chisqTest(c1, c2, c3, c4)
+    Annotation(res._1.orNull, res._2.orNull)
+  },
+    """ Calculates p-value (Chi-square approximation) and odds ratio for 2x2 table """,
+    "c1" -> "value for cell 1", "c2" -> "value for cell 2", "c3" -> "value for cell 3", "c4" -> "value for cell 4"
+  )
+
+  registerAnn("ctTest", chisqStruct, { (c1: Int, c2: Int, c3: Int, c4: Int, minCellCount: Int) =>
```

Comment:
ct = contingency table
Still no?
It's testing exactly that and then its a choice of method for testing that ct....  
This could be extended to arbitrary sized contingency tables also.